extensions/data-workspace/src/main.ts

```diff
@@ -17,9 +17,9 @@ import { ProjectDashboard } from './dialogs/projectDashboard';
 export async function activate(context: vscode.ExtensionContext): Promise<IExtension> {
 	const workspaceService = new WorkspaceService(context);
 	await workspaceService.loadTempProjects();
```

Comment:
Do we have an idea of how long this will take? I wonder if it'd be better to not await this either and instead call checkForProjectsNotAddedToWorkspace in a .then after loadTempProjects completes. 

The potential issue is if this takes a while to read the projects then some of the functionality might be broken until it finishes. But that might be preferable if those actions should be waiting for the initial load to be finished before executing.

Reply:
loadTempProjects shouldn't take too long - it's only adding one project at the most. Sure that makes sense to put it in a .then so the dependency is more clear